src/cpp/handler/controlrequest.cpp

```diff
@@ -37,11 +37,11 @@ class ConnCheck : public Deferred
 	Connection finishedConnection;
 
 public:
-	ConnCheck(ZrpcManager *controlClient, const CidSet &cids, QObject *parent = 0) :
-		Deferred(parent)
+	ConnCheck(ZrpcManager *controlClient, const CidSet &cids) :
+		Deferred()
 	{
-		ZrpcRequest *req = new ZrpcRequest(controlClient, this);
-		finishedConnection = req->finished.connect(boost::bind(&ConnCheck::req_finished, this, req));
+		auto req = std::make_unique<ZrpcRequest>(controlClient, this);
```

Comment:
Will need to save this in a member to avoid it getting destroyed at the end of this function.

Should also remove passing `this` as the parent, since its lifetime will be tracked with the `unique_ptr`.